File: 538_ConvertBSTtoGreaterTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    val = 0
    def convertBST(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root:
            if root.right: self.convertBST(root.right)
            self.val = root.val = self.val + root.val
            if root.left: self.convertBST(root.left)
            return root

File: C++/538_ConvertBSTtoGreaterTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int val = 0;
    TreeNode* convertBST(TreeNode* root) {
        if (root){
            if(root->right){
                convertBST(root->right);
            }
            val = val + root->val;
            root->val = val;
            if(root->left){
                convertBST(root->left);
            }
        }
        return root;
    }
};